# HG changeset patch
# User Rik <rik@octave.org>
# Date 1474323828 25200
#      Mon Sep 19 15:23:48 2016 -0700
# Node ID 77b50a83ec113009e02930e898f3dddde5b6a0ce
# Parent  1e007e986396e01131d531bacb7c363cf6f20390
Error out, rather than segfault, when fontname is "" (bug #49102).

* ft-text-renderer.cc (set_mode): Call ::error rather than ::warning when
bounding box is not set correctly (due to bad fontname).

diff --git a/libinterp/corefcn/ft-text-renderer.cc b/libinterp/corefcn/ft-text-renderer.cc
--- a/libinterp/corefcn/ft-text-renderer.cc
+++ b/libinterp/corefcn/ft-text-renderer.cc
@@ -665,17 +665,17 @@ namespace octave
         bbox = Matrix (1, 4, 0.0);
         line_bbox.clear ();
         push_new_line ();
         break;
 
       case MODE_RENDER:
         if (bbox.numel () != 4)
           {
-            ::warning ("ft_text_renderer: invalid bounding box, cannot render");
+            ::error ("ft_text_renderer: invalid bounding box, cannot render");
 
             xoffset = line_yoffset = yoffset = 0;
             pixels = uint8NDArray ();
           }
         else
           {
             dim_vector d (4, octave_idx_type (bbox(2)),
                           octave_idx_type (bbox(3)));
